src/Raven.Client/Documents/Session/ClusterTransactionSession.cs

```diff
@@ -390,19 +393,29 @@ internal ICommandData GetCommand(DocumentConventions conventions, JsonOperationC
                         var entity = CompareExchangeValueBlittableJsonConverter.ConvertToBlittable(_value.Value, conventions, context, jsonSerializer);
                         var entityJson = entity as BlittableJsonReaderObject;
                         BlittableJsonReaderObject metadata = null;
+                        _originalValue?.Value.TryGet(Constants.Documents.Metadata.Key, out metadata);
+                        var metadataHasChanged = false;
                         if (_value.HasMetadata && _value.Metadata.Count != 0)
                         {
-                            metadata = PrepareMetadataForPut(_key, _value.Metadata, conventions, context);
+                            if (metadata == null)
+                            {
+                                metadataHasChanged = true;
+                                metadata = PrepareMetadataForPut(_key, _value.Metadata, conventions, context); //create new metadata (because there wasn't any metadata before)
```

Comment:
inside `PrepareMetadataForPut` you have some validation that might be relevant.

Reply:
Done